test/e2e/versions_test.go

```diff
@@ -0,0 +1,40 @@
+package e2e_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/test/e2e"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersionParsing(t *testing.T) {
+	versionStr := "v1.3.0\nv1.1.0\nv1.2.0-rc0"
+	versions := e2e.ParseVersions(versionStr)
+	require.Len(t, versions, 3)
+	require.Len(t, versions.FilterOutReleaseCandidates(), 2)
+	require.Equal(t, versions.GetLatest(), e2e.Version{1, 3, 0, false, 0})
+}
+
+// Test case with multiple major versions and filtering out a single major version
+func TestFilterMajorVersions(t *testing.T) {
+	versionStr := "v2.0.0\nv1.1.0\nv2.1.0-rc0\nv1.2.0\nv2.2.0\nv1.3.0"
+	versions := e2e.ParseVersions(versionStr)
+	require.Len(t, versions, 6)
+	require.Len(t, versions.FilterMajor(1), 3)
+}
+
+// Test case to check the Order function
```

Comment:
I think I identified a bug in the `Order` function based on this unit test:

```go
func TestOrderOfReleaseCandidates(t *testing.T) {
	versionsStr := "v1.0.0\nv1.0.0-rc0\nv1.0.0-rc1\n"
	versions := e2e.ParseVersions(versionsStr)
	versions.Order()
	assert.Equal(t, versions[0], e2e.Version{1, 0, 0, true, 0})
	assert.Equal(t, versions[1], e2e.Version{1, 0, 0, true, 1})
	assert.Equal(t, versions[2], e2e.Version{1, 0, 0, false, 0})
}
```

There is a special case that I think needs to be handled. The sorted order should be:
1. v1.0.0-rc0
2. v1.0.0-rc1
3. v1.0.0

but currently v1.0.0 gets sorted before the release candidates

Reply:
Ok nice, let me fix that